clientapi/routing/createroom.go

```diff
@@ -230,44 +325,41 @@ func createRoom(
 		if aliasResp.RoomID != "" {
 			return util.MessageResponse(400, "Alias already exists")
 		}
-	}
 
-	membershipContent := gomatrixserverlib.MemberContent{
-		Membership:  gomatrixserverlib.Join,
-		DisplayName: profile.DisplayName,
-		AvatarURL:   profile.AvatarURL,
+		aliasEvent = &fledglingEvent{
+			Type: gomatrixserverlib.MRoomCanonicalAlias,
+			Content: eventutil.CanonicalAlias{
+				Alias: roomAlias,
+			},
+		}
 	}
 
-	var joinRules, historyVisibility string
-	switch r.Preset {
-	case presetPrivateChat:
-		joinRules = gomatrixserverlib.Invite
-		historyVisibility = historyVisibilityShared
-	case presetTrustedPrivateChat:
-		joinRules = gomatrixserverlib.Invite
-		historyVisibility = historyVisibilityShared
-		// TODO If trusted_private_chat, all invitees are given the same power level as the room creator.
-	case presetPublicChat:
-		joinRules = gomatrixserverlib.Public
-		historyVisibility = historyVisibilityShared
-	default:
-		// Default room rules, r.Preset was previously checked for valid values so
-		// only a request with no preset should end up here.
-		joinRules = gomatrixserverlib.Invite
-		historyVisibility = historyVisibilityShared
-	}
+	var initialStateEvents []fledglingEvent
+	for i := range r.InitialState {
+		switch r.InitialState[i].Type {
```

Comment:
It's not enough to do this, we need to check that `r.InitialState.StateKey == ""` as well, else append them to the list.

Reply:
Good catch, added.